### PR TITLE
Dev: make grow_simulator ==   grow_optmizer

### PR DESCRIPTION
**Pull request by KobusVanRooyen**

### CODE REVIEW

File: src/rtctools_heat_network/workflows/grow_workflow.py [hunk 1]
```diff
@@ -155,85 +156,9 @@ def read(self):
...
-                        f"{demand}.target_heat_demand", ensemble_member
-                    ).values
-                except KeyError:
-                    continue
-                if total_demand is None:
-                    total_demand = demand_values
-                else:
-                    total_demand += demand_values
-                max_demand = max(demand_values)
-                self.__heat_demand_nominal[f"{demand}.Heat_demand"] = max_demand
```
Comment (jimrojerTNO) [inline src/rtctools_heat_network/workflows/grow_workflow.py:178]: can you keep line 158-178 in for the grow workflow, this is needed for scaling.

File: src/rtctools_heat_network/workflows/utils/adapt_profiles.py [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+import numpy as np
+
+from rtctools.data.storage import DataStore
+
+logger = logging.getLogger("WarmingUP-MPC")
+logger.setLevel(logging.INFO)
+
+
+def adapt_hourly_profile_to_common(problem, problem_day_steps):
```
Comment (jimrojerTNO) [inline src/rtctools_heat_network/workflows/utils/adapt_profiles.py:12]: I want the function name to be slightly more descriptive. Can we change the name of this function to: adapt_hourly_year_profile_to_day_averaged_with_hourly_peak_day()

File: src/rtctools_heat_network/workflows/simulator_workflow.py [hunk 3]
```diff
@@ -324,55 +326,73 @@ def times(self, variable=None) -> np.ndarray:
...
+        self.__indx_max_peak, _ = adapt_hourly_profile_to_common(self, self.__day_steps)
+        logger.info("HeatProblem read")
+
+    def _set_data_with_averages_and_peak_day(
+        self,
+        datastore: DataStore,
+        variable_name: str,
+        ensemble_member: int,
+        new_date_times: np.array,
+    ):
```
Comment (jimrojerTNO) [inline src/rtctools_heat_network/workflows/simulator_workflow.py:357]: I would expect this function to also be in the adapt_profiles.py, as it is generic for both workflows. Was there a reason not to do that?

File: src/rtctools_heat_network/workflows/utils/adapt_profiles.py (hunk 2 above)
Comment (KobusVanRooyen) [reply src/rtctools_heat_network/workflows/utils/adapt_profiles.py:12]: Agree

File: src/rtctools_heat_network/workflows/simulator_workflow.py (hunk 3 above)
Comment (KobusVanRooyen) [reply src/rtctools_heat_network/workflows/simulator_workflow.py:357]: My mistake

File: src/rtctools_heat_network/workflows/utils/adapt_profiles.py (hunk 2 above)
Comment (KobusVanRooyen) [reply src/rtctools_heat_network/workflows/utils/adapt_profiles.py:12]: Fixed

File: src/rtctools_heat_network/workflows/simulator_workflow.py (hunk 3 above)
Comment (KobusVanRooyen) [reply src/rtctools_heat_network/workflows/simulator_workflow.py:357]: Fixed

File: src/rtctools_heat_network/workflows/grow_workflow.py (hunk 1 above)
Comment (jimrojerTNO) [reply src/rtctools_heat_network/workflows/grow_workflow.py:178]: missed that, check

File: src/rtctools_heat_network/workflows/utils/adapt_profiles.py [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+        - problem_indx_max_peak: index of the maximum of the peak values
+        - heat_demand_nominal: max demand value found for a specific heating demand
+    """
+
+    def set_data_with_averages_and_peak_day(
+        datastore: DataStore,
+        variable_name: str,
+        ensemble_member: int,
+        new_date_times: np.array,
+    ):
```
Comment (jimrojerTNO) [inline src/rtctools_heat_network/workflows/utils/adapt_profiles.py:27]: Why is this function inside the adapt... function?

Comment (KobusVanRooyen) [reply src/rtctools_heat_network/workflows/utils/adapt_profiles.py:27]: Sorry. My bad

**Review by jimrojerTNO — APPROVED** [review]

File: src/rtctools_heat_network/workflows/utils/adapt_profiles.py [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+                    f"{demand}.target_heat_demand", ensemble_member
+                ).values
+            except KeyError:
+                continue
+            if total_demand is None:
+                total_demand = demand_values
+            else:
+                total_demand += demand_values
+            heat_demand_nominal[f"{demand}.Heat_demand"] = max(demand_values)
+            heat_demand_nominal[f"{demand}.Heat_flow"] = max(demand_values)
```
Comment (KobusVanRooyen) [inline src/rtctools_heat_network/workflows/utils/adapt_profiles.py:92]: @jimrojerTNO: this line has been added to be the same as master, which originally comes from the branch "Bug fix inconsistent ub (#36)"

**Pull request merged**